### PR TITLE
finished assessment

### PR DESCRIPTION
**Pull request by KevinBurgin**

### CODE REVIEW

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Can you uncomment the code you want me to review? please and thank you!

File: code-challenges.test.js [hunk 1]
```diff
@@ -18,49 +21,126 @@
...
-
+// create a function called shuffle
+const shuffle = (array) => {
+// initiate a variable set to nothing
+    let m;
+    let ray;
+// loop over array
+    for(let i = array.length -1; i>0; i--){
+// use math.floor and math. random to randomize array
+        m = Math.floor(Math.random() * (i +1))
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:46]: Whats the thought process with this logic?

File: code-challenges.test.js [hunk 2]
```diff
@@ -18,49 +21,126 @@
...
+
+// tried a different approach
+const minMax = (arr) => {
+    let min = Math.min(...nums1)
+    let max = Math.max(...nums1)
+    let newArr = []
+}
+console.log(minMax(nums1))
+
+// same as first but for min
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:102]: I love seeing "INFINITY"-- How can you put these two functions together?

File: code_challenges.rb [hunk 3]
```diff
@@ -22,6 +34,25 @@
 album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
 
+# create a function only_vowels
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:37]: This logic is super cool!

File: code_challenges.rb [hunk 4]
```diff
@@ -30,11 +61,57 @@
...
+        end
+    end
+    def get_info 
+        "#{@model}, #{@wheels}, #{@current_speed}"
+    end
+end
+
+new_bike = Bike.new('Haro', 2, nil)
+
+p new_bike.brake
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:112]: did you test other methods here? looks good!

File: interview-questions.md [hunk 5]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer:Every block in ruby returns a value automatically of the last executed code in the block this is because of Implicit return.
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Ruby is an object oriented program that uses classes and objects.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:34]: I can see where you're going with this answer and I appreciate the attempt to answer with an example. Next time try answering the big 6 questions: Who, What, When, Where, Why, How? That should give you a few sentences of info to elaborate on.

**Review by WalkerTexas-coder — COMMENTED** [review]
Great work Kevin! Interesting code great communication. Thank you for making the code clean and readable. Keep up the good work!  

I left a few comments for you to reply to. Once you have feel free to merge.

File: code-challenges.test.js (hunk 1 above)
Comment (KevinBurgin) [reply code-challenges.test.js:46]: honestly I felt like i was close so i just kept plugging in stuff that might work and it got messy so a lot of it was me just testing things

File: code-challenges.test.js (hunk 2 above)
Comment (KevinBurgin) [reply code-challenges.test.js:102]: I could use an empty array and push the values into it but for some reason coulndt get it to work

File: code_challenges.rb (hunk 4 above)
Comment (KevinBurgin) [reply code_challenges.rb:112]: yea it worked for me

File: interview-questions.md (hunk 5 above)
Comment (KevinBurgin) [reply interview-questions.md:34]: ok sounds good thanks

**Pull request merged**